### ISSUE TITLE
update_project_provisioning unable to parse development provisioning profile

### ISSUE THREAD

**Issue by youssef-meherzi** [fastlane/fastlane#11796]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When using update_project_provisioning to set my development provisioning profile, it crashes because the script wasn't able to parse the provisioning profile.

/Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/update_project_provisioning.rb:33:in `initialize': [!] Could not parse the PKCS7: nested asn1 error (ArgumentError)

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
[15:36:47]: ------------------
[15:36:47]: --- Step: sigh ---
[15:36:47]: ------------------

+-------------------------------------+-----------------------------------+
|                         Summary for sigh 2.80.0                         |
+-------------------------------------+-----------------------------------+
| development                         | true                              |
| app_identifier                      | com.xxxxxx.ios.dev            |
| provisioning_name                   | OB-Dev Dev                        |
| filename                            | build/OB-Dev.mobileprovision      |
| readonly                            | true                              |
| skip_install                        | true                              |
| skip_certificate_verification       | true                              |
| adhoc                               | false                             |
| force                               | false                             |
| username                            | xxxxxx@gmail.com |
| team_id                             | xxxxxx                        |
| ignore_profiles_with_different_name | false                             |
| skip_fetch_profiles                 | false                             |
| platform                            | ios                               |
+-------------------------------------+-----------------------------------+

[15:36:48]: Starting login with user 'xxxxxx@gmail.com'
[15:36:49]: Successfully logged in
[15:36:49]: Fetching profiles...
[15:36:50]: Found 1 matching profile(s)
[15:36:50]: Downloading provisioning profile...
[15:36:51]: Successfully downloaded provisioning profile...
/Users/ci/git/oslo-front/mobile/xxxxx-app-mobile-ios/build/OB-Dev.mobileprovision
[15:36:51]: Setting Provisioning Profile type to 'development'
[15:36:51]: -----------------------------------------
[15:36:51]: --- Step: update_project_provisioning ---
[15:36:51]: -----------------------------------------
[15:36:51]: You’re updating provisioning profiles directly in your project, but have you considered easier ways to do code signing?
[15:36:51]: https://docs.fastlane.tools/codesigning/GettingStarted/
[15:36:51]: Parsing mobile provisioning profile from 'build/OB-Dev.mobileprovision'
[15:36:52]: -------------------
[15:36:52]: --- Step: slack ---
[15:36:52]: -------------------
[15:36:53]: Successfully sent Slack notification
+--------------------+--------------------------------------------------------------------------------------------+
|                                                  Lane Context                                                   |
+--------------------+--------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM   | ios                                                                                        |
| PLATFORM_NAME      | ios                                                                                        |
| LANE_NAME          | ios functionalTests                                                                        |
| SIGH_PROFILE_PATH  | /Users/ci/git/oslo-front/mobile/xxxxxx-app-mobile-ios/build/OB-Dev.mobileprovision     |
| SIGH_PROFILE_PATHS | ["/Users/ci/git/oslo-front/mobile/xxxxxx-app-mobile-ios/build/OB-Dev.mobileprovision"] |
| SIGH_PROFILE_TYPE  | development                                                                                |
+--------------------+--------------------------------------------------------------------------------------------+
[15:36:53]: Could not parse the PKCS7: nested asn1 error

+------+--------------------------------+-------------+
|                  fastlane summary                   |
+------+--------------------------------+-------------+
| Step | Action                         | Time (in s) |
+------+--------------------------------+-------------+
| 1    | Verifying fastlane version     | 0           |
| 2    | default_platform               | 0           |
| 3    | cocoapods                      | 6           |
| 4    | Switch to ios build lane       | 0           |
| 5    | Switch to ios strip_certs lane | 0           |
| 6    | set_info_plist_value           | 0           |
| 7    | set_info_plist_value           | 0           |
| 8    | sigh                           | 3           |
| 💥   | update_project_provisioning    | 0           |
| 10   | slack                          | 0           |
+------+--------------------------------+-------------+

[15:36:53]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
/Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/update_project_provisioning.rb:33:in `initialize': [!] Could not parse the PKCS7: nested asn1 error (ArgumentError)
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/update_project_provisioning.rb:33:in `new'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/update_project_provisioning.rb:33:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:231:in `block in execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
	from Fastfile:73:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:201:in `try_switch_to_lane'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:156:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
	from Fastfile:109:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/commands_generator.rb:337:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/cli_tools_distributor.rb:98:in `take_off'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:22:in `load'
	from /usr/local/bin/fastlane:22:in `<main>'
</pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.13.3                                                        |
| Ruby                        | 2.3.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.14.3 (Apple Git-98)                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.13.3 (17D47)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.2.7                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 9.2                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | fr_FR.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.13.0"
fastlane_require 'json'
default_platform :ios

import "../../CommonFastfile"

platform :ios do
  before_all do
    cocoapods
  end

  desc "Submits all Env Build to Apple TestFlight"
  lane :launch do |options|
    env = options[:env]
    buildNumber = options[:buildNumber]
    versionNumber = options[:versionNumber]
    
    array = env.split(',')
    array.each{
      |x| puts x.to_s
      build env:x.to_s, signature:"release", versionNumber:versionNumber, buildNumber:buildNumber
      if options[:deploy]
        deploy env:x.to_s
      end
    }
  end

  desc "Build an application locally"
  lane :build do |options|
    env = options[:env]
    config = env.capitalize
    strip_certs env:env, bb:"../WebAssets/assets/backbase"
    
    set_info_plist_value(path: "CXPMobile/Info.plist", key: "CFBundleShortVersionString", value: options[:versionNumber])
    set_info_plist_value(path: "CXPMobile/Info.plist", key: "CFBundleVersion", value: options[:buildNumber])
    
    signature = options[:signature]
    if signature == "release"
      development = false
      provisioningName = 'OB-'+config+' Dist'
      codesigningIdentity = 'iPhone Distribution: xxxxxx (xxxxxx)'
    end
    if signature == "debug"
      development = true
      provisioningName = 'OB-'+config+' Dev'
      codesigningIdentity = 'iPhone Developer: xxxxxx (xxxxxx)'
    end
    raise 'Skipping build because of un unknown signature type : '+signature+'!' unless provisioningName != nil
      
    provisionPath = 'build/OB-'+config+'.mobileprovision'
    sigh(
      development: development,
      app_identifier: appIdentifier(env),
      provisioning_name: provisioningName,
      filename: provisionPath,
      readonly: true,
      skip_install: true,
      skip_certificate_verification: true
    )
    update_project_provisioning(
      xcodeproj: "xxxxxx.xcodeproj",
      profile: provisionPath,
      target_filter: "xxxxxx",
      build_configuration: config
    )
    gym(
      workspace: "xxxxxx.xcworkspace",
      configuration:config,
      scheme: "xxxxxx",
      output_directory: "Build/",
      output_name: "xxxxxx.ipa",
      silent: false,
      clean: true,
      codesigning_identity: codesigningIdentity,
      xcargs: 'UPLOAD_DSYM="1"',
      skip_profile_detection: true
    )
  end

  desc "Submits a new Env Build to Apple TestFlight"
  lane :deploy do |options|
    env = options[:env]

    pilot(
      skip_submission: true,
      skip_waiting_for_build_processing: true,
      app_identifier: appIdentifier(env)
    )
  end

  desc "Launch functional tests"
  lane :functionalTests do |options|
    env = options[:env]
    buildNumber = options[:buildNumber]
    versionNumber = options[:versionNumber]
    build env:env, signature:"debug", versionNumber:versionNumber, buildNumber:buildNumber
    
    sh("ps", "|", "grep", "[a]ppium", "|", "awk", "'{print", "$1}'", "|", "xargs", "kill", "-9")
    sh("appium", "&")
    Dir.chdir("..") do
      sh("npm", "install")
      sh("npm", "test")
    end
  end

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
     slack(
       message: exception.message,
       success: false
     )
  end
    
  def appIdentifier(env)
    app_identifier = 'com.xxxxxx.ios'
    if env != 'pro'
      app_identifier << '.'+env 
    end
    return app_identifier
  end
end

# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.xxxxxx.ios" # The bundle identifier of your app
apple_id "xxxxxx@gmail.com" # Your Apple email address

team_id "xxxxxx"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.80.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 2.3.2        |
| CFPropertyList            | 2.3.6        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.60.0       |
| faraday                   | 0.14.0       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| io-console                | 0.4.5        |
| libxml-ruby               | 2.9.0        |
</details>

*generated on:* **2018-02-07**
</details>

**Comment by youssef-meherzi** [comment]
I've found that : 

- The openssl command used in the lane fail also on the command line on that provisioning profile but succeed in the other provisioning profiles.
- The one that doesn't work is a development one that contain 32 certificates and more than 100 device.
- When I reduce the size of my provisioning by deselecting most of them in the member center, it works again.
I deduced that openssl have some trouble decoding bigger files.

What I did as a workaround : parsing the provisioning profile and pass the UUID directly to the build command
`%x[security cms -D -i ../#{provisionPath} > ../#{provisionPath}.plist]
    provisionUUID = get_info_plist_value(path: "#{provisionPath}.plist", key: "UUID")`
